packages/core/src/components/chip/index.js

```diff
@@ -0,0 +1,68 @@
+class Chip {
+  static instances = new WeakMap();
+
+  static get cssClasses() {
+    return {};
+  }
+
+  static get strings() {
+    return {
+      INIT_SELECTOR: '.ray-chip'
+    };
+  }
+
+  static create(element, options) {
+    return this.instances.get(element) || new this(element, options);
+  }
+
+  static createAll(target = document, _options = { active: false }) {
+    // Finds all instances of select on the document or within a given element and instantiates them.
+    const options = {
+      initSelector: this.strings.INIT_SELECTOR,
+      ..._options
+    };
+
+    const chips = Array.from(target.querySelectorAll(options.initSelector));
+    chips.forEach(select => this.create(select, options));
+  }
+
+  constructor(root, options) {
+    this._root = root;
+
+    this.constructor.instances.set(this._root, this);
+
+    this.state = {
+      active: options.active
+    };
+
+    this._bindEventListeners();
+    this.assignClasses();
+  }
+
+  _bindEventListeners() {
+    this._root.addEventListener('mousedown', this.onMousedown);
```

Comment:
Thoughts on adding a `touchstart` event as well.. not sure what the best UX is for this - seeing on my phone that they don't work very well for quicker, slightly clumsier gestures